oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -877,35 +877,94 @@ impl super::Node for WasmNode {
-        "test_function".to_string(),
-        reader_handle,
-    );
-
-    // The minimal module does not have the necessary entry point, so it should fail immediately.
-    assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    use wat::{parse_file, parse_str};
+
+    fn setup_node<S: AsRef<[u8]>>(buffer: S, entrypoint: &str) -> Box<dyn Node> {
+        let configuration = crate::runtime::Configuration {
+            nodes: HashMap::new(),
+            entry_module: "test_module".to_string(),
+            entrypoint: entrypoint.to_string(),
+        };
+        let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
+        let (_, reader_handle) =
+            runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
+
+        let module = wasmi::Module::from_buffer(buffer).unwrap();
+
+        Box::new(WasmNode::new(
+            "test",
+            runtime_ref,
+            TEST_NODE_ID,
+            Arc::new(module),
+            entrypoint.to_string(),
+            reader_handle,
+        ))
+    }
+
+    fn from_testfile(path: &str) -> Vec<u8> {
+        parse_file(path).unwrap()
+    }
+
+    #[test]
+    fn wasm_starting_module_without_content_fails() {
+        // Loads an empty module, equivalent to: vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00,
+        // 0x00] The minimal module does not have the necessary entry point, so it should
+        // fail immediately. From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
+        // Minimal module:
+        // \0asm - magic
+        // 0x01 - version (in little-endian)
+        let binary = from_testfile("../../testdata/empty.wat");
+        let mut node = setup_node(binary, "oak_main");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_minimal_module_succeeds() {
+        let binary = from_testfile("../../testdata/minimal.wat");
+        let mut node = setup_node(binary, "oak_main");
+        assert_eq!(Ok(()), node.start());
+        node.stop();
+    }
+
+    #[test]
+    fn wasm_starting_module_missing_an_export_fails() {
+        let binary = from_testfile("../../testdata/missing.wat");
+        let mut node = setup_node(binary, "oak_main");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_module_with_wrong_export_fails() {
+        let binary = from_testfile("../../testdata/minimal.wat");
+        let mut node = setup_node(binary, "oak_other_main");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_module_with_wrong_signature_fails() {
+        let binary = from_testfile("../../testdata/wrong.wat");
+        let mut node = setup_node(binary, "oak_main");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_module_with_no_channels_succeeds() {
```

Comment:
Yes I'd like to keep just one around, just in case anyone in the future feels inspired to do something along those lines, it would be nice to have this as a starting point. Thanks!